source/shared/cpp/ObjectModel/SharedAdaptiveCard.cpp

```diff
@@ -85,6 +89,17 @@ std::shared_ptr<AdaptiveCard> AdaptiveCard::Deserialize(
         ParseUtil::GetString(json, AdaptiveCardSchemaKey::BackgroundImage);
     std::string speak = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Speak);
     ContainerStyle style = ParseUtil::GetEnumValue<ContainerStyle>(json, AdaptiveCardSchemaKey::Style, ContainerStyle::None, ContainerStyleFromString);
+    std::string language = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Language, true /* isRequired */);
+    
+    std::locale locale;
+    try
+    {
+        locale = std::locale(language);
```

Comment:
what happens when language string is not set? i think we should continue rendering using device's system setting. i think we have facility to propagate warnings for case like this and we could it.
Have you tested the changes on other platforms? language string is not standardized, so "en-US" may work with windows but may not work with iOS.  Correct string could be "en_US" in iOS.

Reply:
in fact this works If the string passed is "en" (english) or "es" (spanish), I'm not sure if the same behavior works on Linux

Reply:
but that does remind me that I forgot to update the unit tests

Reply:
We have not fully speced that lang tag support. Namely, one of the things that remains to be speced is the format of the tag. What we do not want is to be in the business of mapping a set of language strings to platform-specific languages, on each platform. We should wait until the spec is ready before we implement anything IMO.

Reply:
what other platforms did you test beside windows? can locale object be instantiated using language tag alone?

---
In reply to: [166160065](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r166160065) [](ancestors = 166160065)

Reply:
@matthidinger